A small fix in do_bench_using_profiling

y-sq opened the pull request:
Summary: Results: https://docs.google.com/document/d/1B_4rtiDFPH_jV3VpnqLPnInwDMpF7yX29G82UoJTcu8/edit?tab=t.0

Test Plan:
```
buck2 run mode/opt  -c fbcode.enable_gpu_sections=true ai_acceleration/float8/benchmarks/bench:bench_fp8_shapes_eval 2>&1 | tee output44.txt
```

Rollback Plan:

Differential Revision: D76298690




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

y-sq commented:
@pytorchbot  label "topic: not user facing"

nmacchioni commented on torch/_inductor/utils.py:
```diff
@@ -296,6 +296,8 @@ def do_bench_using_profiling(
     for _ in range(n_warmup):
         fn()
 
+    torch.cuda.synchronize()
```
Hey! Wondering what the basis for this change is?

I'm not an expert on `do_bench_using_profiling`, but for the standard implementation `do_bench` we actually don't want a synchronization here for a couple of reasons. One reason is that the warmups actually "pad" the event queue so that our first few iterations in the actual timing loop aren't in cpu-gpu lockstep; in addition, the gpu might have time to clock down during the synchronization which would partially eliminate the warmup's benefits.

If there's a difference in `do_bench_using_profiling` that makes this beneficial I'd be super interested to hear! I could imagine that it might be messing up the profiling run somehow. I checked your linked doc but didn't find context there, I might be looking in the wrong place. Thanks!

y-sq replied:
When using `do_bench_using_profiling`, I sometimes got the error from: https://github.com/pytorch/pytorch/blob/a728f95b580615a52af14f0d2e4db0d6a00816c5/torch/_inductor/utils.py#L327

I was suspecting it might be because the missing of `sync` after warmup, and the profiler is capturing some unfinished kernel calls from warmup. Adding this `sync` fixed my issue.


> One reason is that the warmups actually "pad" the event queue so that our first few iterations in the actual timing loop aren't in cpu-gpu lockstep;

I think this won't be an issue in `do_bench_using_profiling`. It calculates the latency from all GPU events in the profiler result, 
https://github.com/pytorch/pytorch/blob/a728f95b580615a52af14f0d2e4db0d6a00816c5/torch/_inductor/utils.py#L301
https://github.com/pytorch/pytorch/blob/a728f95b580615a52af14f0d2e4db0d6a00816c5/torch/_inductor/utils.py#L322


> the gpu might have time to clock down during the synchronization which would partially eliminate the warmup's benefits.

However, I don't have a good answer for this. If this makes the result much more different, I'll revert my changes here.

y-sq committed:
[float8] Eval Benchmark (#155500)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/155500

Results: https://docs.google.com/document/d/1B_4rtiDFPH_jV3VpnqLPnInwDMpF7yX29G82UoJTcu8/edit?tab=t.0

Test Plan:
```
buck2 run mode/opt  -c fbcode.enable_gpu_sections=true ai_acceleration/float8/benchmarks/bench:bench_fp8_shapes_eval 2>&1 | tee output44.txt
```

Rollback Plan:

Reviewed By: yoyoyocmu, nmacchioni

Differential Revision: D76298690
